Merge 'Hosted Onboarding' in to Mollie Connect documentation (#726)
mollie/api-documentation: source/oauth/onboarding.rst

Onboard your customers at Mollie
================================
This guide will help you with the process of getting your customers onboard at Mollie. From the first OAuth permission
to sending data about your customer, all steps will be covered. It also covers our Hosted Onboarding feature.

We are using the concept of a new customer but you can use the Onboarding APIs also for existing merchants that have
already an account at Mollie.

.. note:: In this guide we assume that you have already created an OAuth app in the Mollie Dashboard. If you do not know
          how to do this, please see the :doc:`Getting started guide </oauth/getting-started>`.

Prerequisites
-------------
If you have access to our Hosted Onboarding-feature and want to use this, you need to `turn on hosted onboarding
<https://www.mollie.com/dashboard/settings/hosted-onboarding>`_ in the Mollie Dashboard.

You will have to provide a logo, name and redirect URL for the OAuth app that will be automatically created.

> That page will display the Client ID and secret needed for the steps below.

Reviewer: ```suggestion
After submitting, you will receive the Client ID and secret needed for the steps below.
```

Step 1: Setting up the authorization
------------------------------------

You will need to acquire authorization for the

> Mollie account your customer creates or logs into. This :doc:`page </oauth/overview>` gives you an overview of the flow. If you just want to follow the implementation steps, follow this guide: :doc:`Mollie Connect getting started </oauth/getting-started>`.

Reviewer: ```suggestion
Mollie organization your customer creates or logs in to. This :doc:`page </oauth/overview>` gives you an overview of the flow. If you just want to follow the implementation steps, follow this guide: :doc:`Mollie Connect getting started </oauth/getting-started>`.
```

>   .. note::  Make sure to use the Client Id and secret provided of the Hosted Onboarding APP if you want to use this feature. These can be found in the `hosted onboarding settings <https://www.mollie.com/dashboard/settings/hosted-onboarding>`_.

Reviewer: ```suggestion
  .. note::  Make sure to use the Client ID and secret displayed in the `hosted onboarding settings <https://www.mollie.com/dashboard/settings/hosted-onboarding>`_ if you want to make use of this feature.
```

Require the following scopes to board merchants and create payments on their behalf:

.. list-table::
   :widths: auto

   * - | ``payments.write``
       | Payments API
     - Create payments for the merchant. The received payment will be added to the merchant's balance.

   * - | ``organizations.write``
       | Organizations API
     - Change the merchant's organizational details.

   * - | ``onboarding.read``
       | Onboarding API
     - View the merchant's onboarding status.

   * - | ``onboarding.write``
       | Onboarding API
     - Submit onboarding data for the merchant.

> Step 2: Customer signs up / logs in and gives authorization

Reviewer: ```suggestion
Step 2: Merchant signs up or logs in, and authorizes access to their organization
```

Reviewer: @arnetm We usually try to avoid the term "Customer", since it creates confusion with the Customers API. I do see the existing OAuth Onboarding page already uses this term. What are your thoughts on this?

Author: I would go with this change.

In that situation you will want to redirect them straight to the onboarding URL instead of them landing in the platform. **This is only possible if Hosted Onboarding is activated.** The Hosted Onboarding URL is received in the Get Onboarding Status call.

On the other hand, if they need to start the onboarding in Mollie but didn't just authorize, you can just show them the default "need-data" view that we will detail in the next step.

To make your customer experience easier, you can help them onboard at Mollie by prefilling some fields using the :doc:`Submit onboarding data endpoint </reference/v2/onboarding-api/submit-onboarding-data>`. This can happen before you redirect the customer towards the Hosted Onboarding URL.

Step 5: Customer needs to complete the onboarding
-------------------------------------------------
The :doc:`Onboarding status </reference/v2/onboarding-api/get-onboarding-status>` response is used to display a proper message and action to your customer.

> There's two milestones that the customer reaches during the Mollie onboarding, ``canReceivePayments`` (basic information has been provided) and ``canReceiveSettlements`` (all information has been provided and verified).

Reviewer: ```suggestion
There's two milestones that the customer reaches during the Mollie onboarding, ``canReceivePayments`` and ``canReceiveSettlements``.
```

You won't always receive `canReceivePayments: true` after providing basic information. The consumer should make no assumptions about this.

Author: good one 👍
